package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -620,19 +661,51 @@ def _read_next_timestep(self):
             except IndexError:
                 raise ValueError("No coordinate information detected")
         elif not self.lammps_coordinate_convention in convention_to_col_ix:
-            raise ValueError(f"No coordinates following convention {self.lammps_coordinate_convention} found in timestep")
+            raise ValueError("No coordinates following convention"
+                             + f"{self.lammps_coordinate_convention}"
+                             + "found in timestep")
 
         coord_cols = convention_to_col_ix[self.lammps_coordinate_convention]
 
         ids = "id" in attr_to_col_ix
+
+        # Create the data arrays for additional attributes which will be saved
+        # under ts.data
+        additional_keys = []
+        if len(attrs) > 3:
+            for attribute_key in attrs:
+                # Skip the normal columns
+                if (attribute_key == "id" or
+                        attribute_key in
+                        self._coordtype_column_names[
+                            self.lammps_coordinate_convention
+                        ] or attribute_key not in self._additional_columns):
+                    continue
+                # Else this is an additional field
+                ts.data[attribute_key] = np.empty(self.n_atoms)
```

Comment:
Are we allocating a new array *for every new timestep*? 

If yes, is there any way around doing this, e.g., allocate the entries in init or at least only allocate once??

Reply:
I suppose you could allocate once  and have a check like 

```python
if not ts.data[attribute_key]:
    ts.data[attribute_key] = np.empty(self.n_atoms)
else:
    ts.data[attribute_key][:] = blah
```

Reply:
If the columns no longer exist in following time steps, would the entry be nonexistent in the next timestep (since not parsed on the next read) or contain the old data? LAMMPS would in principle allow shenanigans I think.

Reply:
I think this was the right resolution as lammps allows all kinds of stuff.